src/util/http_utils.js

```diff
@@ -664,6 +665,82 @@ function set_amz_headers(req, res) {
     res.setHeader('x-amz-id-2', req.request_id);
 }
 
+const s3_error_options = {
+    ErrorClass: S3Error,
+    error_missing_content_length: S3Error.MissingContentLength
+};
+/**
+ * @param {Object} req
+ * @param {http.ServerResponse} res
+ */
+async function set_expiration_header(req, res) {
+    if (req.method === 'HEAD' || req.method === 'GET' || req.method === 'PUT') {
+        const rules = req.params.bucket && await req.object_sdk.read_bucket_lifecycle_config_info(req.params.bucket);
+        const object_md = {
+                bucket: req.params.bucket,
+                key: req.params.key,
+                size: req.headers['x-amz-decoded-content-length'] || req.headers['content-length'] ? parse_content_length(req, s3_error_options) : undefined,
+                tagging: req.body && req.body.Tagging ? s3_utils.parse_body_tagging_xml(req) : undefined,
```

Comment:
Could you explain why you assigned this `object_md`?

Reply:
Updated, Thanks